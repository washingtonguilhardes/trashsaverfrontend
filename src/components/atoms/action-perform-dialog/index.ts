export * from './action-perform-dialog.atom';
