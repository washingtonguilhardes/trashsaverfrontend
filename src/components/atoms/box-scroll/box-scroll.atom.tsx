import styled from '@emotion/styled';

import Box from '@mui/material/Box';

export const BoxScroll = styled(Box)<{ outerHeight?: string }>`
  height: calc(100vh - ${({ outerHeight = '198px' }) => outerHeight});
  overflow: auto;
  &::-webkit-scrollbar {
    width: 16px;
  }

  &::-webkit-scrollbar-track {
    background-color: #e4e4e4;
    border-radius: 100px;
  }

  &::-webkit-scrollbar-thumb {
    border-radius: 100px;
    background-image: linear-gradient(180deg, #009be5 0%, #708ad4 99%);
    box-shadow: inset 2px 2px 5px 0 rgba(#fff, 0.5);
  }
`;
