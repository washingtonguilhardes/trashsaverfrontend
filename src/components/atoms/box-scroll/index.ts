export * from './box-scroll.atom';
