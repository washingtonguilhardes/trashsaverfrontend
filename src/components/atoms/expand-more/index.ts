export * from './expand-more.component';
