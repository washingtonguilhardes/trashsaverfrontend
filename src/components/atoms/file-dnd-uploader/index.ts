export * from './file-dnd-uploader.atom';
