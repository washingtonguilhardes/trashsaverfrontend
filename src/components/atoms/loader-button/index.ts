export * from './loader-button.atom';
