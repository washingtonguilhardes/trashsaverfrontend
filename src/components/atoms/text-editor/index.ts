export * from './text-editor.atom';
