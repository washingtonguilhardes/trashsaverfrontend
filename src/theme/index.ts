import { createTheme } from '@mui/material/styles';

// Create a theme instance.
const defaultTheme = createTheme({
  palette: {
    primary: {
      light: '#63ccff',
      main: '#009be5',
      dark: '#101F33',
    },
  },
  typography: {
    h5: {
      fontWeight: 500,
      fontSize: 26,
      letterSpacing: 0.5,
    },
  },
  shape: {
    borderRadius: 8,
  },
  components: {
    MuiTab: {
      defaultProps: {
        disableRipple: true,
      },
    },
  },
  mixins: {
    toolbar: {
      minHeight: 48,
    },
  },
});

export const theme = {
  ...defaultTheme,
  components: {
    MuiDrawer: {
      styleOverrides: {
        paper: {
          backgroundColor: '#081627',
        },
      },
    },
    MuiButton: {
      styleOverrides: {
        root: {
          textTransform: 'none',
        },
        contained: {
          boxShadow: 'none',
          '&:active': {
            boxShadow: 'none',
          },
        },
      },
    },
    MuiTabs: {
      styleOverrides: {
        root: {
          marginLeft: defaultTheme.spacing(1),
        },
        indicator: {
          height: 3,
          borderTopLeftRadius: 3,
          borderTopRightRadius: 3,
          backgroundColor: defaultTheme.palette.common.white,
        },
      },
    },
    MuiTab: {
      styleOverrides: {
        root: {
          textTransform: 'none',
          margin: '0 16px',
          minWidth: 0,
          padding: 0,
          [defaultTheme.breakpoints.up('md')]: {
            padding: 0,
            minWidth: 0,
          },
        },
      },
    },
    MuiIconButton: {
      styleOverrides: {
        root: {
          padding: defaultTheme.spacing(1),
        },
      },
    },
    MuiTooltip: {
      styleOverrides: {
        tooltip: {
          borderRadius: 4,
        },
      },
    },
    MuiDivider: {
      styleOverrides: {
        root: {
          backgroundColor: 'rgb(255,255,255,0.15)',
        },
      },
    },
    MuiListItemButton: {
      styleOverrides: {
        root: {
          '&.Mui-selected': {
            color: '#4fc3f7',
          },
        },
      },
    },
    MuiListItemText: {
      styleOverrides: {
        primary: {
          fontSize: 14,
          fontWeight: defaultTheme.typography.fontWeightMedium,
        },
      },
    },
    MuiListItemIcon: {
      styleOverrides: {
        root: {
          color: 'inherit',
          minWidth: 'auto',
          marginRight: defaultTheme.spacing(2),
          '& svg': {
            fontSize: 20,
          },
        },
      },
    },
    MuiAvatar: {
      styleOverrides: {
        root: {
          width: 32,
          height: 32,
        },
      },
    },
  },
};
