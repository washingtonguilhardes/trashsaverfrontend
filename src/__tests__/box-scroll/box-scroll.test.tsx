import { ThemeProvider } from '@emotion/react';
import { BoxScroll } from '@src/components/atoms/box-scroll';
import { theme } from '@src/theme';
import { render, screen } from '@testing-library/react';

describe('<BoxScroll />', () => {
  test('expect render', () => {
    render(
      <ThemeProvider theme={theme}>
        <BoxScroll data-testid="element">A child text</BoxScroll>
      </ThemeProvider>
    );
    expect(screen.getByTestId('element')).toMatchSnapshot();
  });
});
