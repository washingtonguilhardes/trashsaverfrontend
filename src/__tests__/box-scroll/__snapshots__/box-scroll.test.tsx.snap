// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BoxScroll /> expect render 1`] = `
<div
  class="MuiBox-root css-prcuy6"
  data-testid="element"
>
  A child text
</div>
`;
